plugins/csv/tests/output.rs

```diff
@@ -0,0 +1,276 @@
+use std::fs::{self};
+use std::time::{Duration, UNIX_EPOCH};
+
+use alumet::{
+    agent::{
+        self,
+        plugin::{PluginInfo, PluginSet},
+    },
+    measurement::{MeasurementBuffer, MeasurementPoint, Timestamp, WrappedMeasurementValue},
+    metrics::RawMetricId,
+    pipeline::naming::OutputName,
+    plugin::PluginMetadata,
+    resources::{Resource, ResourceConsumer},
+    test::{RuntimeExpectations, runtime::OutputCheckInputContext},
+    units::Unit,
+};
+use indoc::indoc;
+
+use plugin_csv::{Config, CsvPlugin};
+use tempfile;
+
+pub const TIMEOUT: Duration = Duration::from_secs(10);
+
+fn config_to_toml_table(config: &Config) -> toml::Table {
+    toml::Value::try_from(config).unwrap().as_table().unwrap().clone()
+}
+
+struct TestMetrics {
+    metric_u64: RawMetricId,
+    metric_f64: RawMetricId,
+}
+
+impl TestMetrics {
+    pub fn get(ctx: &OutputCheckInputContext) -> Self {
+        Self {
+            metric_u64: ctx.metrics().by_name("test_metric_u64").expect("metric should exist").0,
+            metric_f64: ctx.metrics().by_name("test_metric_f64").expect("metric should exist").0,
+        }
+    }
+}
+
+fn simple_point(metric: RawMetricId, value: WrappedMeasurementValue) -> MeasurementPoint {
+    MeasurementPoint::new_untyped(
+        Timestamp::from(UNIX_EPOCH),
+        metric,
+        Resource::LocalMachine,
+        ResourceConsumer::LocalMachine,
+        value,
+    )
+}
+
+#[test]
+fn csv_output() {
+    let _ = env_logger::Builder::from_default_env().try_init();
+
+    // Prepare the plugin
+
+    let tmp = tempfile::tempdir().unwrap();
+    let file_path = tmp.path().join("alumet-output.csv");
+
+    let result_file: String = file_path.clone().into_os_string().into_string().unwrap();
+    let result_file_2 = result_file.clone();
```

Comment:
You don't need a string to use `std::fs::read_to_string`. You can keep a `Path` or `PathBuf`.